litrpc/netcmds.go

```diff
@@ -64,6 +65,25 @@ func (r *LitRPC) Connect(args ConnectArgs, reply *StatusReply) error {
 	return nil
 }
 
+// ------------------------- name a connection
+type AssignNicknameArgs struct {
+	Peer     uint32
+	Nickname string
+}
+
+func (r *LitRPC) AssignNickname(args AssignNicknameArgs, reply *StatusReply) error {
+	err := r.Node.SaveNicknameForPeerIdx(args.Nickname, args.Peer)
+	if err != nil {
+		return err
+	}
+
+	r.Node.RemoteCons[args.Peer].Nickname = args.Nickname
```

Comment:
This can crash if the peer number doesn't exist.
First do
```
peer, ok := r.Node.RemoteCons[args.Peer]
if !ok{
return fmt.Errorf("peer %d doesn't exist", args.Peer)
}
```